src/evo/netinfo.h

```diff
@@ -42,6 +44,91 @@ constexpr std::string_view NISToString(const NetInfoStatus code)
     assert(false);
 }
 
+class NetInfoEntry
+{
+public:
+    enum NetInfoType : uint8_t {
```

Comment:
enum class here too?

Reply:
Not kept as an `enum class` as `m_type` is a serialized value and could be set to anything during deserialization. Making it an `enum class` would result in either

```c++
private:
    uint8_t m_type{static_cast<uint8_t>(NetInfoType::Invalid)};
```

or (assuming `m_type` is a `NetInfoType`)

```c++
    template <typename Stream>
    void Unserialize(Stream& s)
    {
        s >> static_cast<uint8_t>(m_type);
```

which now provides a false sense of security that `NetInfoType` only holds a recognized value when it could be any `uint8_t`.